src/pages/home/report/ReportActionItemFragment.js

```diff
@@ -1,38 +1,51 @@
 import React, {memo} from 'react';
-import {ActivityIndicator, View} from 'react-native';
 import PropTypes from 'prop-types';
-import Str from 'expensify-common/lib/str';
 import reportActionFragmentPropTypes from './reportActionFragmentPropTypes';
 import styles from '../../../styles/styles';
-import variables from '../../../styles/variables';
-import themeColors from '../../../styles/themes/default';
-import RenderHTML from '../../../components/RenderHTML';
 import Text from '../../../components/Text';
-import * as EmojiUtils from '../../../libs/EmojiUtils';
-import withWindowDimensions, {windowDimensionsPropTypes} from '../../../components/withWindowDimensions';
 import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
-import * as DeviceCapabilities from '../../../libs/DeviceCapabilities';
 import compose from '../../../libs/compose';
-import convertToLTR from '../../../libs/convertToLTR';
 import {withNetwork} from '../../../components/OnyxProvider';
 import CONST from '../../../CONST';
-import applyStrikethrough from '../../../components/HTMLEngineProvider/applyStrikethrough';
-import editedLabelStyles from '../../../styles/editedLabelStyles';
 import UserDetailsTooltip from '../../../components/UserDetailsTooltip';
 import avatarPropTypes from '../../../components/avatarPropTypes';
+import * as ReportUtils from '../../../libs/ReportUtils';
+import AttachmentCommentFragment from "./comment/AttachmentCommentFragment";
+import TextCommentFragment from "./comment/TextCommentFragment";
+import RenderCommentHTML from "./comment/RenderCommentHTML";
 
 const propTypes = {
-    /** Users accountID */
-    accountID: PropTypes.number.isRequired,
-
     /** The message fragment needing to be displayed */
     fragment: reportActionFragmentPropTypes.isRequired,
 
-    /** Is this fragment an attachment? */
-    isAttachment: PropTypes.bool,
+    /** Additional styles to add after local styles. */
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]).isRequired,
+
+    /** localization props */
+    ...withLocalizePropTypes,
+
+    // Properties required for a COMMENT fragment
+
+    /** The reportAction's source */
+    source: PropTypes.oneOf(['Chronos', 'email', 'ios', 'android', 'web', 'email', '']),
+
+    /** The reportAction's pending action */
+    pendingAction: PropTypes.oneOf(['add', 'update', 'delete']),
 
-    /** If this fragment is attachment than has info? */
-    attachmentInfo: PropTypes.shape({
+    /** Does the reportAction have a comment thread? */
+    hasCommentThread: PropTypes.bool.isRequired,
+
+    // Properties required for an attachment COMMENT fragment
+
+    /** Is this fragment a part of a reportAction that has not yet loaded? */
+    isLoading: PropTypes.bool.isRequired,
+
+    // TODO: Is it actually only related to the uploaded attachments?
+    /** Is this fragment a part of an uploaded attachment? */
+    isUploadedAttachment: PropTypes.bool,
```

Comment:
> Uploading attachment... is considered an attachment

I know, and for now, I think we can consider that intended...

> isUploadedAttachment is a bit misleading because currently we pass action.isAttachment to it

It was my intention to name `action.isAttachment`'s contract better. As we know, that is a local-only property for attachments that were uploaded from this very client.

If we agree to nuke [the path that makes use](https://github.com/Expensify/App/pull/25524#discussion_r1299182331) of this property, this discussion would become outdated. That's what I would prefer.

Reply:
Sure, I'll try to find someone from the internal team to carefully review this PR and do some confirmations on the backend side